americangut/diversity_analysis.py

```diff
@@ -0,0 +1,1729 @@
+
+    # Identifies the center positions for the p text
+    p_cents = [(centers[1] - centers[0])/2. + centers[0]]
+
+    if len(centers) > 2:
+        for idx, center in enumerate(centers[2:]):
+            p_cents.append((center - centers[0])/2. + centers[0])
+    # Determines the p text
+    p_text = []
+    for p in p_values:
+        if p < 0.01 and show_value:
+            p_text.append('%1.e' % p)
+        elif show_value:
+            p_text.append('%1.2f' % p)
+        elif p < 0.0001:
+            p_text.append('****')
+        elif p < 0.001:
+            p_text.append('***')
+        elif p < 0.01:
+            p_text.append('**')
+        elif p < 0.05:
+            p_text.append('*')
+        elif p < 0.1:
+            p_text.append('+')
+        else:
+            p_text.append('')
+
+    lines = []
+
+    # Plots the first comparison
+    l1 = []
+    l1.append(ax.plot([centers[0]]*2, [tops[0] + space, bar_levels[-1]], 'k-'))
+    l1.append(ax.plot([centers[1]]*2, [tops[1] + space, bar_levels[0]], 'k-'))
+    l1.append(ax.plot([centers[0], centers[1]], [bar_levels[0]]*2, 'k-'))
+    l1.append(ax.text(x=p_cents[0], y=bar_levels[0] + 0.2*space, s=p_text[0],
+                      size=label_size, ha='center'))
+    lines.append(l1)
+
+    if len(centers) > 2:
+        for idx, p_cent in enumerate(p_cents[:-1]):
+            ln = []
+            ln.append(ax.plot([centers[idx + 2]]*2,
+                              [tops[idx + 2] + space, bar_levels[idx + 1]],
+                              'k-'))
+            ln.append(ax.plot([centers[0], centers[idx + 2]],
+                              [bar_levels[idx + 1]]*2, 'k-'))
+            ln.append(ax.text(x=p_cents[idx + 1],
+                              y=bar_levels[idx + 1] + 0.2*space,
+                              s=p_text[idx+1], size=label_size, ha='center'))
+            lines.append(ln)
+
+    return lines
+
+
+def segment_colormap(cm_name, n_colors, n_pad=None, start=None):
+    """Segments a matplotlib colormap into descrete colors
+
+    Parameters
+    ----------
+    cm_name : str
+        The name of the continous matplotlib colormap which will be segmented.
+    n_colors : int
+        The number of colors needed in the colormap
+    n_pad : int, optional
+        The number of total colors to have in the colormap. By default, this
+        will be one more than the number of colors provided.
+    start : int, optional
+        The number of colors to skip over before display starts. By default,
+        this is `n_pad` - `n_colors`.
+
+    Returns
+    -------
+    new_map : array
+        A segmented array containing the colormap
+
+    """
+
+    # Sets parameters if necessary
+    if n_pad is None:
+        n_pad = n_colors + 1
+    if start is None:
+        start = n_pad - n_colors
+
+    # Gets the colormap
+    cm = mpl.cm.get_cmap(cm_name)
+    # Sets up the new map
+    new_map = np.array([list(cm(1. * (i + start) / n_pad)) for i in
```

Comment:
why cast to `list`? this is essentially an `np.array` of a `list` of `list`s, or is the return from a `Colormap` not numpy friendly?